array-manipulation-3/src/zip.js

```diff
@@ -0,0 +1,31 @@
+/* exported zip */
+// assign an empty array into two variables named array and output
+// write a for loop that iterates through the first array
+// if the length of first is equal to the second, push the value of first[i] and second[i] into array
+// push the value of array into output and then clear the value of array
+// if the length of the first and second are not equal, check if any of the values of second[i] are not equal to undefined
+// if it is not equal to undefined, push the value of first[i] and second[i] into array
+// push the value of array into output and then clear the value of array
+// return the variable output
+
+function zip(first, second) {
+  var output = [];
+  var array = [];
+
+  for (var i = 0; i < first.length; i++) {
+    if (first.length === second.length) {
+      array.push(first[i]);
+      array.push(second[i]);
+      output.push(array);
+      array = [];
+    } else {
+      if (second[i] !== undefined) {
```

Comment:
Rather than do all this, I would just find the shorter length before you begin the loop and use that as the length to loop until.